sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -326,12 +327,40 @@ internal static void CopyHeaders(HttpHeaders from, HttpHeaders to)
 #if NET6_0_OR_GREATER
         private static string JoinHeaderValues(HeaderStringValues values)
         {
-            return values.Count switch
-            {
-                0 => string.Empty,
-                1 => values.ToString(),
-                _ => string.Join(",", values)
-            };
+            var count = values.Count;
+            // Explicitly handle most common cases
+            switch (count) {
+                case 1:
+                    return values.ToString();
+                case 0:
+                    return string.Empty;
+                case 2:
+                    using (var enumerator = values.GetEnumerator())
```

Comment:
This is a very common case. Unfortunately, the only way to access elements of the `HeaderStringValues` is to use enumerator.

Reply:
`DefaultInterpolatedStringHandler` is also good